### PR TITLE
Add weather app logic and the UI

### PR DESCRIPTION
**Pull request by descholar-ceo**
## What does this PR do?
It Adds the whole app logic and the design

## Tasks to be completed on this PR
- [x] Write the whole logic of the app
- [x] Add some styling
- [x] Polish codes
- [x] Fix linters errors
- [x] Add professional readme file

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1 +1,75 @@
...
+
+### Prerequisites
+- [NodeJS](https://nodejs.org/)
+- [Yarn](https://yarnpkg.com/) or just the default npm which comes with NodeJS
+
+### Setup
+- Run `git clone https://github.com/descholar-ceo/weather-app && cd weather-app` to get a copy of source codes on your local computer and to navigate inside the project directory
+- Run `yarn install`
+- Run `yarn build`
+- Create a file called `.env` add `PORT` in the same format of `.env.example`
```
Comment (adaorachi) [inline README.md]: - [x] In this instruction, it is not very clear on what should be done here. Kindly add that the port you want the webpack server to open the app on should be set here. e.g - `PORT=8080`. In the `.env.example`, what you have specified is just this - `PORT=`, and this is not well understood by non-techy users.

File: README.md [hunk 2]
```diff
@@ -1 +1,75 @@
...
+- Webpack
+
+## Features
+  - Search the weather info of any city on the earth
+  - Convert the gotten result from degrees celcius to fahreneit
+
+## Deployment
+- [Wearher app on netlify](https://weather-app-descholar.netlify.app)
+- [Wearher app on heroku](https://weather-app-descholar.heroku.com)
+
```
Comment (adaorachi) [inline README.md]: - [x] Kindly correct the typo here. It should be `Weather app` instead.

**Review by adaorachi — CHANGES_REQUESTED** [review]
### Hi @descholar-ceo :wave: 
You've done a good job on this project. Your weather app works as it should and the features are well-implemented. :+1: 
You met almost all requirements for this project except minor changes to be done on your README file.
Kindly see the highlighted code below.

---
### STATUS: CHANGES REQUIRED

---

- [**optional**] It will be nice if you implement a ‘loading’ component that displays from the time the form is submitted until the information comes back from the API ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/javascript/weather_app/loading_component_requirement.png) from Odin)

 - [**optional**] It will be nice if your app fetches the current geolocation of the user when the app is first accessed and fetches the weather from that location

**After implementing the MANDATORY changes, you can request for another code review**

Should you have any question, kindly reach out to me on Slack with the username @adaorachi

Happy Coding :relaxed: 

MaryAnn Chukwuka

File: README.md (hunk 2 above)
Comment (descholar-ceo) [reply README.md]: Resolved

File: README.md (hunk 1 above)
Comment (descholar-ceo) [reply README.md]: Resolved

**Comment by descholar-ceo** [comment]
> ### Hi @descholar-ceo 
> You've done a good job on this project. Your weather app works as it should and the features are well-implemented. 
> You met almost all requirements for this project except minor changes to be done on your README file.
> Kindly see the highlighted code below.
> 
> ### STATUS: CHANGES REQUIRED
> * [**optional**] It will be nice if you implement a ‘loading’ component that displays from the time the form is submitted until the information comes back from the API ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/javascript/weather_app/loading_component_requirement.png) from Odin)
> * [**optional**] It will be nice if your app fetches the current geolocation of the user when the app is first accessed and fetches the weather from that location
> 
> **After implementing the MANDATORY changes, you can request for another code review**
> 
> Should you have any question, kindly reach out to me on Slack with the username @adaorachi
> 
> Happy Coding 
> 
> MaryAnn Chukwuka

@adaorachi Thank you for the review, about the optional `loading component` I think it's already there, the scond optional `fetching user's geolocation` I have opened [this issue](https://github.com/descholar-ceo/weather-app/issues/2), I will implement as soon as possible. Thank you!

**Review by adaorachi — APPROVED** [review]
### Hi @descholar-ceo :wave: 
Well done implementing the changes requested in the previous review. Your setup instruction looks well detailed now. Good Job :100:  
You've met all the minimum requirements for this project, so it's a :+1:  for me.
You can merge this branch to the base branch of your project while you start working on the next project.

---

### STATUS: PROJECT APPROVED :100: 

---

Happy Coding and Good luck :relaxed: 

MaryAnn Chukwuka

**Pull request merged**